duplicate_url_discarder/processors/base.py

```diff
@@ -1,10 +1,10 @@
 from abc import ABC, abstractmethod
-from typing import Any, Tuple
+from typing import Any, Optional, Tuple
 
 
 class UrlProcessorBase(ABC):
-    def __init__(self, args: Tuple[Any, ...]):
-        self.args: Tuple[Any, ...] = args
+    def __init__(self, args: Optional[Tuple[Any, ...]] = None):
```

Comment:
Currently `or ()` is done when creating `UrlRule` so this change only matters when creating a processor instance directly, but that indeed can be useful I think.

Reply:
Yeah I was thinking that the _Learners_ will have direct access to the processors later on and thus, it should also handle cases like `None`.